service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/ServiceGenerator.java

```diff
@@ -133,6 +134,12 @@ private void addInterfaceFields(ServiceIDL serviceIdl, JCodeModel jCodeModel, JD
         invocation.arg(JExpr.lit(serviceIdl.getVersion().getMinor()));
         invocation.arg(JExpr.lit(serviceIdl.getVersion().getPatch()));
         versionField.init(invocation);
+
+        if (serviceIdl.getService().getObjectMapperClass() != null) {
+            fieldModifier = JMod.PUBLIC | JMod.STATIC | JMod.FINAL;
```

Comment:
Can this be protected and non static? Basically generated service clients are singleton. We don't need to make this "public static final". Instead "protected final" should work